[17.0][FIX] account_payment, website_payment: Respect provider website_id in multi-website scenarios

xaviedoanhduy opened the pull request:
In a multi-website environment, payment providers are often configured specifically for each website using the `website_id` field. While the checkout page (`/shop/payment`) correctly filters providers by the current website, other routes such as `/my/payment_method` or `/payment/pay` flows do not apply this filtering consistently.

This patch ensures that the `website_id` constraint on payment providers is respected across all relevant flows, improving consistency and preventing users from seeing or using providers that are not available for their current website.

Without this patch, users may see or select payment providers that are not intended for their site, leading to potential confusion, incorrect transactions, or access to providers that are not supported on the current website.


---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

anso-odoo committed:
[FIX] web: remove `Dialog` header bg

Before this commit: header color of `Dialog` on mobile was "community
color" or white, depending on the version of Odoo the database is
running. This was creating issues in in the front-end, where the user of
the website would see back-end specific colors in different `Dialog`s.

This commit removes the color customization of `Dialog`'header as it
was fitting a design line we had before we introduced MILK.

task-4001365

closes odoo/odoo#170087

Signed-off-by: Pierre Paridans (app) <app@odoo.com>

visp-odoo committed:
[FIX] web_editor: fix sidebar overflow issue

Steps to Reproduce:

- Set the Odoo backend language to Spanish.
- Enter edit mode.
- Drag and drop the "Text-Image" snippet.
- Click on the image within the dropped snippet.
Bug: The "Layout" menu in the sidebar overflows, causing a horizontal
scrollbar.

Issue:
The width of the `we-button` elements within the `we-button-group`
was not handled correctly. In the Spanish translation, the text in the
"Layout" options was too long, resulting in overflow and a horizontal
scrollbar in the sidebar.

Solution:
This commit addresses the layout issue specifically for DOM structure
involving `we-button-group` and `we-select`. Without affecting other
styles, it applies a grid template to equally divide the available space
among the buttons, preventing overflow.

task-4378522

closes odoo/odoo#217722

X-original-commit: 23d8df67d92af678531b050b598e227fc6c3b540
Signed-off-by: Benjamin Vray (bvr) <bvr@odoo.com>

d-fence committed:
[IMP] tests: allow to add test_tags at runtime

This commit add the possibility to automatically add test-tag on a test
method at runtime. A generic method `get_method_additional_tags` is
added on `BaseCase` test class. That method can be overridden to return
a list of test-tags that will be added on the test methods.

With this mechanism, the `HttpCase` class override this method to
add a `is_tour` test-tag when the `start_tour` method is used in the
test method. Also, the `start_tour` method will now emit a warning when
the method is called without being tagged `is_tour`. That way, all the
tours can now be started with the `is_tour` test-tag.

Part-of: odoo/odoo#217102
Related: odoo/enterprise#89934
Signed-off-by: Xavier Dollé (xdo) <xdo@odoo.com>
Signed-off-by: Xavier Morel (xmo) <xmo@odoo.com>

d-fence committed:
[FIX] mail, test_main_flows: add missing `is_tour` test-tag

Following the new test-tags feature that allows to add additional
test-tags at runtime, the test methods that are calling `start_tour` are
tagged `is_tour`. On the other hand, when the `start_tour` method is
called outside of a test method, a warning is emitted.

This commit fixes some tests that are using such an intermediate method
to start a tour by manually adding the `is_tour` test-tag.

Part-of: odoo/odoo#217102
Related: odoo/enterprise#89934
Signed-off-by: Xavier Dollé (xdo) <xdo@odoo.com>
Signed-off-by: Xavier Morel (xmo) <xmo@odoo.com>

d-fence committed:
[IMP] tests: add an automatic is_query_count test-tag

With this commit, a `is_query_count` test-tag is automatically added on
tests methods that use an assertQueryCount by overriding the
`get_method_additional_tags`.

Part-of: odoo/odoo#217102
Related: odoo/enterprise#89934
Signed-off-by: Xavier Dollé (xdo) <xdo@odoo.com>
Signed-off-by: Xavier Morel (xmo) <xmo@odoo.com>

d-fence committed:
[FIX] google_calendar, website*: add missing `is_query_count` test-tag

Following the new test-tags feature that allows to add additional
test-tags at runtime, the test methods that are using `assertQueryCount`
are tagged `is_query_count`. On the other hand, when the `assertion`
method is called outside of a test method, a warning is emitted.

This commit fixes tests which hide the assertion by manually adding
the `is_query_count` test-tag.

Part-of: odoo/odoo#217102
Related: odoo/enterprise#89934
Signed-off-by: Xavier Dollé (xdo) <xdo@odoo.com>
Signed-off-by: Xavier Morel (xmo) <xmo@odoo.com>

aych-odoo committed:
[FIX] mail: hide message actions on transient messages

Before this commit, the toggle-star and mark-as-read actions were visible
on chatbot messages in a non-persisted livechat thread.
Clicking these actions caused errors.

This commit hides these actions when the message is in non-persisted state,
preventing such errors.

task-4743758

closes odoo/odoo#214382

Signed-off-by: Alexandre Kühn (aku) <aku@odoo.com>

dwa-odoo committed:
[FIX] loyalty: prevent archiving pricelists linked to active loyalty programs

Step to Reproduce:
1. Install `sale_loyalty` and `sale`
2. Enable the `Pricelists` option in the settings.
3. Create a pricelist named demo.
4. Create a Discount & Loyalty named `test program` with type `Discount Code`,
5. Assign the demo pricelist to the loyalty program.
6. Copy the discount code from the program’s conditional rules.
7. Archive the demo pricelist.
8. create sale order for any product.
9. Try to apply the copied coupon code.

Observation:
- An error is shown: "This code is invalid".
- check loyalty program `test program` pricelist field is empty, that suggesting
  it's valid for all pricelists, but the coupon still fails.

Issue:
- When a linked pricelist is archived, it's hidden in the UI,
  but the relation still exists in the relational table.
- While filtering the domain for coupon, the condition is not satisfied due to
  the program's pricelist.

solution:
- Display a validation error to the user when trying to archive a pricelist
  that is linked to any active promotional programs.

opw-4841678

closes odoo/odoo#214968

Signed-off-by: Dhwani Patel (dwa) <dwa@odoo.com>

Commit:
[FIX] stock: Prevent the overlapping of rows in reception reports

## Short functional explanation of the error
When printing a reception report containing a lot of products, the top row
of the pages after the first one is overlapped with its corresponding
header in the resulting PDF.

## Reproduction Steps
1. Open the settings. In the inventory section, enable "Reception Report".
2. Create a new quotation and add at least 40 different products. Each
product should have at least 1 copy in stock, and the quantity to order must
be greater than the quantity we have of this product in stock.
3. Confirm the sales order.
4. Create a new purchase order and add the exact same products you added in
the quotation.
5. Confirm the order and click on the "Recept" smart button.
6. Click on the "Allocation" smart button and click on the "Assign all"
gray button.
7. Click on print and open the PDF once it finishes downloading.

### Expected behavior
Each row (corresponding to the reception of a product) and its
corresponding header is printed properly,  regardless of how many pages
constitute the report.

### Unexpected behavior
The PDF has more than one page, and on the top of the second page, the
product row is overlapped with the header of its corresponding section.

## Origin of the issue
It comes from an issue with WKHtmltopdf itself:
https://github.com/wkhtmltopdf/wkhtmltopdf/issues/1524
The report uses the "thead" tag for the headers. However, in some cases,
when the header is linked to a table dynamically (with for-each, for
 example), WebKit ignores or breaks the "Thead" behavior.

## Explanation of the fix
I override the default behavior of thead, which repeats the header
automatically at each new page, with a new style. Now, the "Thead"
special behavior is ignored and treated like ordinary lines. The header
isn't shown at each new page anymore, avoiding the overlapping of product
rows.

closes odoo/odoo#215398

Signed-off-by: William Henrotin (whe) <whe@odoo.com>

mathcoutant committed:
[FIX] account_edi_ubl_cii: use `deferred_start_date` and `deferred_end_date` in FacturX XML

Issue:
The FacturX XML has a fiel `BillingSpecifiedPeriod`, it should depend on `deferred_start_date` and `deferred_end_date` if they are there. Currently it only outputs the invoice date as start date and the due date as end date.

Steps to reproduce:
- Install "l10n_de" and switch to a German company
- Go to a contact, in the page Accounting > Electronic Invoicing change the format to "Factur-X (CII)"
- Create an invoice for this contact
- Add a start date and end date on the line of this invoice (`deferred_start/end_date`)
- Confirm and send to Factur-X
- In the generated XML the dates used are the invoice date and the due date

Cause:
As the `deferred_start_date` and `deferred_end_date` fields are defined in enterprise and the XML generation is in community, this [commit](https://github.com/odoo/odoo/commit/f0c5d5b46444a289bf55ad5846623d48ac3a3b71) set dates defined in comunity instead of the deferred ones.

Solution:
Mimicking [the way it's done for UBL20](https://github.com/odoo/odoo/blob/638268a81ed5a292a02d7fc353c4954159de54e1/addons/account_edi_ubl_cii/models/account_edi_xml_ubl_20.py#L846) we check if  `deferred_start_date` and `deferred_end_date` are defined (i.e. account_accountant is installed).
If it's the case we use the min/max of the `deferred_start_date` and `deferred_end_date` as start/end date of the `BillingSpecifiedPeriod` in the XML.

To include as much info as we can this commit also adds the `deferred_start_date` and `deferred_end_date` on lines.

opw-4874370

closes odoo/odoo#215595

Signed-off-by: Claire Bretton (clbr) <clbr@odoo.com>

DjamelTouati committed:
[FIX] quality_control: switch to MTS when move line to failure_location

Steps to reproduce the bug:
Steps to reproduce the bug:
- Unarchive the MTO route
- Create a product P1:
    - Type: Storable
    - Route: MTO + Buy
    - Supplier: Vendor A
- Create a quality point:
    - Product: P1
    - Control per: Quantity
    - Failure location: WH/input/order Processing

- Create a sales order for 1 unit of P1
- Confirm the SO
    → A purchase order is created for Vendor A with 1 unit of P1
- Confirm the PO
- Go to the related PO picking
- Fail the quality check

Problem:
picking related to the SO remains in the "Waiting Availability" state
and the stock move is not switched to the Make to stock procure method

opw-4874199

closes odoo/odoo#216337

Related: odoo/enterprise#88791
Signed-off-by: William Henrotin (whe) <whe@odoo.com>

antonrom1 committed:
[FIX] website_hr_recruitment: email search special chars

Searches failed to return expected results when the query included special characters

task-4935570

suma-odoo committed:
[FIX] l10n_it_edi_ndd: prevent wrong value error

The computation of `l10n_it_document_type` fails when multiple
`l10n_it.document.type` records share the same code.
This can happen if a user duplicates an existing Document Type or creates a new
one with the same code, causing `get()` on the grouped recordset to return
multiple results.
This fix adds a check to ensure that the `code` field remains
unique across all `l10n_it.document.type` records.

Steps to Reproduce:

- Install the `l10n_it_edi_ndd` module.
- create a new invoice and confirm it.
- from `Electronic Invoicing` page, Open Document Type and duplicate it.
- Create another invoice and confirm it.

observation: you will receive a traceback for wrong value error

opw - 4902513

closes odoo/odoo#217155

Signed-off-by: Wala Gauthier (gawa) <gawa@odoo.com>

hmpb-odoo committed:
[FIX] l10n_mx: Fixing diot tags.

The language used in the DIOT documentation was confusing and even though we had the exact same description for two columns it turns out they were different, so we need to change the logic of a few columns to make it work as needed.
- Behavior prior to this PR: In the DIOT, the columns Paid 8% N, Paid 8% s, Paid 16%, Importation 16% and Intangible imports 16% are sums of taxes paid with this type of tax. And all the tax columns are based in the base not in the tax part.
- Behavior after this PR. Taxes now have a correction in their tags so that the columns in the DIOT report are calculated correctly, Tax columns only contains tax values, not base values.

opw-4920577

"I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr"

closes odoo/odoo#217441

Related: odoo/enterprise#89481
Signed-off-by: Antoine Dupuis (andu) <andu@odoo.com>

mathcoutant committed:
[FIX] account_edi_ubl_cii: fix FacturX validation issues

Issue:
There are several issues with the XML of FacturX.

Steps to reproduce:
- Create a partner with a direct debit mandate
- In the Accounting tab of the partner select "FacturX" as eInvoice format
- Create an invoice with this partner, pay with SDD, confirm
- Send to FacturX
- Verify the PDF with https://www.portinvoice.com/
	- Missing namespace
- In the XML the code for the payment means does not adapt to the payment method: 42 (Payment to bank account) and not 59 (SDD)

Cause:
- The namespaces are not there (It's not mandatory).
- We always input "42" as payment mean

Solution:
- Added namespace `xmlns:qdt="urn:un:unece:uncefact:data:standard:QualifiedDataType:100"` and `xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"`
- Added the code "59" for SDD, as a lot of codes could be added we add a dictionnary that could be used to add codes. This commit only handles the SDD code

opw-4907827

closes odoo/odoo#217763

Signed-off-by: Paolo Gatti (pgi) <pgi@odoo.com>

sben-odoo committed:
[FIX] account: prevent company currency update

To_reproduce:
==============
1- switch company.
2- update any other company currency that has journal items.
3- company currency changed.

Problem:
=========
- When switching companies, users could update the other company's
currency even when journal items existed for that company.
Solution:

=========
- Added .sudo() to the account.move.line search to bypass access rights
and properly detect existing journal items in that company.

opw-4920206

closes odoo/odoo#217988

X-original-commit: ee612ac3e859991becec964f91af8831324e840a
Signed-off-by: Josse Colpaert (jco) <jco@odoo.com>
Signed-off-by: Saif Allah Ben Khalil (sben) <sben@odoo.com>

Commit:
[FIX] base: res.country _name_search arbitrary capitalization

**Current behavior:**
When importing a res.country record, if specifying a `name` then
the search performed expects it to exactly match a
first-letter-only capitalized pattern.

**Expected behavior:**
Let arbitrary capitalization be used on the input name string.

**Steps to reproduce:**
1. Create a sheet to import to `Contacts` (res.partner)
* `name`: anything
* `country_id`: "ARMENIA"

2. Load the sheet for import -> test -> can't resolve country
name

**Cause of the issue:**
The `res.country` records' names follow a
first-letter-capitalized pattern and we do an exact search on
the strings.

**Fix:**
Use `capitalize()` on the input value when we do the search,
which will transform the string into precisely the form we want.

opw-4780334

closes odoo/odoo#216870

Signed-off-by: Chong Wang (cwg) <cwg@odoo.com>

gawa-odoo committed:
[FIX] account_edi_ubl_cii,account_peppol: UBL TR is used for Nilvera

It's not a format that is on peppol and does not use the fields
from peppol.

closes odoo/odoo#218220

Signed-off-by: Florian Gilbert (flg) <flg@odoo.com>

lvsz committed:
[FIX] website_sale: recompute prices on address change

Versions
--------
- 16.0
- 17.0

Fixed in FW starting from 18.0

Steps
-----
1. Have a price included & a price excluded tax;
2. have a fiscal position that maps one unto the other;
3. open a cart on /shop by adding a product with the relevant taxes;
4. check the untaxed amount;
5. in Portal view, change your address to trigger a new fiscal position;
6. check the untaxed amount.

Issue
-----
The taxes have changed, but so has the untaxed amount.

Cause
-----
Commit 39b2bb36df9d6 recomputes fiscal position & taxes on address
change, but doesn't recompute the prices.

Solution
--------
After recomputing the taxes, recompute the prices.

opw-4844132

closes odoo/odoo#218150

X-original-commit: ede8846987a987c5a3c46741cdc2c451e38fcb70
Signed-off-by: Levi Siuzdak <sile@odoo.com>

LouisGobert committed:
[PERF] account: optimize query for company filtering

This fix aims to avoid a subquery in the main query and instead use a
IN clause to improve search performance.

On a database with over 37 million journal items, before the fix, the
query to load the 40 potentials journal items to reconcile in the bank
reconciliation widget took around 50 seconds; after the fix, it take
only 3 seconds (with only 1 company to filter).

Subquery when 'child_of' is used:
```
...
AND (
    "account_move_line"."company_id" IN (
        SELECT
            "res_company"."id"
        FROM
            "res_company"
        WHERE
            ("res_company"."parent_path" LIKE '1/%')
    )
)
```

Query after the fix:
```
...
AND ("account_move_line"."company_id" IN (1))
```

opw-4896863

closes odoo/odoo#218034

Signed-off-by: Wala Gauthier (gawa) <gawa@odoo.com>

xaviedoanhduy committed:
[FIX] {account,website}_payment: filter providers per website

nilshamerlinck commented on addons/website_payment/controllers/__init__.py:
```diff
@@ -2,3 +2,4 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from . import portal
+from . import payment
```
nitpick: respect order

nilshamerlinck reviewed (approved):
pre-approving

Armin-FalDiS committed:
[FIX] base: menu icon image support

PR#135607 breaks support of image types other than
`png` for menu icons.
This puts back the support for previously
supported image types.

closes odoo/odoo#218131

Signed-off-by: Martin Trigaux (mat) <mat@odoo.com>

MaxTheWhale committed:
[FIX] hw_drivers: drivers deleted on bad response

Before this commit, if the IoT box was connected a DB which then went
offline (e.g. expired runbot), and it tried to download drivers, the
request would fail but it would still delete the old handlers.

After this commit, we avoid this problem in two ways. Firstly, we check
that we received an OK HTTP status before trying to process the response
body. Secondly, we check that the reponse we have received is a valid
zip file BEFORE deleting the old handlers. This way we only delete them
if we are sure we have something to replace them with.

task-4933638

closes odoo/odoo#218163

Signed-off-by: Max Whale (mawh) <mawh@odoo.com>

d-fence committed:
[FIX] pos_online_payment, mail, survey, crm, project: add missing test tags

With the new test-tags features that allows to add additional test tags
at runtime, the tests that starts a tour or that are using a query_count
and that are not detected as such must be tagged respectively `is_tour`
or `is_query_count`.

closes odoo/odoo#217102

Related: odoo/enterprise#89934
Signed-off-by: Xavier Dollé (xdo) <xdo@odoo.com>
Signed-off-by: Xavier Morel (xmo) <xmo@odoo.com>

C3POdoo committed:
[I18N] *: fetch 17.0 translations

MaxTheWhale committed:
[FIX] hw_drivers: use lock when printing on windows

Before this commit, if multiple documents were sent to print at exactly
the same time, some of them could end up erroring because the Windows
printing API requires us to send one document at a time.

After this commit, we use a lock similar to the Linux driver, to ensure
that only one document is sent to the printer at once.

opw-4829908

closes odoo/odoo#218300

Related: odoo/enterprise#89930
Signed-off-by: Max Whale (mawh) <mawh@odoo.com>

khsr-odoo committed:
[FIX] base: update state name from 'Orissa' to 'Odisha' as per govt guidelines

Steps to reproduce:
1. Install the `Contacts` module.
2. Go to Contacts > Create a new contact.
3. Select country "India", then open the State dropdown.
4. Observe that "Orissa" appears instead of the updated name "Odisha".

Issue:
As per Government of India guidelines, the state name was officially changed
from "Orissa" to "Odisha" in 2011.
However, Odoo still uses the outdated name in the state selection.

Solution:
Update the name of the state from "Orissa" to "Odisha" in state records.

opw-4935633

closes odoo/odoo#218448

X-original-commit: 4767737c5527d5b6e204f5ecdf0d35979aeb8404
Related: odoo/enterprise#89998
Signed-off-by: Nicolas Lempereur (nle) <nle@odoo.com>
Signed-off-by: Khushi Srivastava (khsr) <khsr@odoo.com>

shrm-odoo committed:
[FIX] website_sale: prevent duplicate category list display on shop page

Steps to reproduce:
1. Go to the shop page.
2. Open the editor and configure Categories to display on the left, and
   Attributes at the top.
3. Save the changes and click the offcanvas toggle button next to the layout
   buttons.

Issue:
- The category list appears twice: once on the left side and again in the
  offcanvas dropdown. This is redundant and affects the user experience.

Cause:
- The category list in the offcanvas menu isn’t restricted to mobile view, so it
  also displays on larger screens where it’s already visible in the sidebar.

Fix:
- Add the 'd-lg-none' class to hide the category list in the offcanvas menu on
  large devices.

opw-4830180

closes odoo/odoo#218330

Signed-off-by: Valentin Chevalier <vcr@odoo.com>

prdu-odoo committed:
[FIX] sale_project_stock: Added post_install to profitability tests

One of the tests failed without demo data due to a lacking stock accounting account. Since the test requires the Chart of Accounts, it was moved to post-install.

runbot error: [163084](https://runbot.odoo.com/odoo/error/163084)

closes odoo/odoo#218346

Signed-off-by: Xavier Bol (xbo) <xbo@odoo.com>

lvsz committed:
[FIX] website_sale: handle translated express checkout partners

Versions
--------
- 17.0+

Steps
-----
1. Change website language to Spanish
2. add a deliverable order to cart;
3. go via express checkout.

Issue
-----
The express checkout partner doesn't get updated as expected.

Cause
-----
Before this commit, it checks whether the express checkout partner's
name ends with the order reference, as is the case in English. In
Spanish however, the order reference gets used in the middle of the
name.
As a consequence, the `_create_or_edit_partner` method does not get
called.

Solution
--------
Check whether the order reference is part of the partner name.

opw-4894059

closes odoo/odoo#217528

Signed-off-by: Valentin Chevalier <vcr@odoo.com>

clbr-odoo committed:
[IMP] account_peppol: Improve crons to run correctly in multiple batches

For database with big invoice volume, it is necessary to be able to
retrieve documents in smaller batches otherwise crons will timeout
due to the 15 minutes limit.
Add a retrigger mechanic to the possibly heavy crons.

opw-4925532

closes odoo/odoo#218383

Signed-off-by: Antoine Dupuis (andu) <andu@odoo.com>

tde-banana-odoo committed:
[FIX] mail: do not reset activity type with wrong model

Just warn models are invalid, but do not reset for stable. Data is not enforced
hence no real issue with wrong models.

Followup of odoo/odoo#156731

closes odoo/odoo#218549

Signed-off-by: Thibault Delavallee (tde) <tde@openerp.com>

Commit:
[FIX] portal_share: fix typo

fix a small typo and and refactor for clarity

opw-4938011

closes odoo/odoo#218600

X-original-commit: f97e1a2fa6628ff1ecbab1c5d0ea9bf8e4dcf42c
Signed-off-by: Florian Vranckx (flvr) <flvr@odoo.com>

LucasLefevre committed:
[FIX] spreadsheet: update o_spreadsheet to latest version

### Contains the following commits:

https://github.com/odoo/o-spreadsheet/commit/8c0d3dae7 [REL] 17.0.69 [Task: 0](https://www.odoo.com/odoo/2328/tasks/0)
https://github.com/odoo/o-spreadsheet/commit/89b7d29f7 [FIX] Composer: Remove formatting when pasting external content [Task: 4910559](https://www.odoo.com/odoo/2328/tasks/4910559)
https://github.com/odoo/o-spreadsheet/commit/8fa50e3b6 [PERF] import xlsx: avoid reading all cells multiple times during table_conversion [Task: 4699860](https://www.odoo.com/odoo/2328/tasks/4699860)
https://github.com/odoo/o-spreadsheet/commit/36c8f2ba9 [FIX] Config: remove eslint [Task: 0](https://www.odoo.com/odoo/2328/tasks/0)
https://github.com/odoo/o-spreadsheet/commit/73a832b86 [IMP] config: ignore .DS_Store file from mac os [](https://www.odoo.com/odoo/2328/tasks/)
https://github.com/odoo/o-spreadsheet/commit/d1f3c7733 [FIX] header_size: resize row based on link label instead of full link [Task: 4886598](https://www.odoo.com/odoo/2328/tasks/4886598)
https://github.com/odoo/o-spreadsheet/commit/c2cf091c8 [FIX] demo: Fix new `clear` action [Task: 0](https://www.odoo.com/odoo/2328/tasks/0)

closes odoo/odoo#218591

Signed-off-by: Vincent Schippefilt (vsc) <vsc@odoo.com>
Co-authored-by: Anthony Hendrickx (anhe) <anhe@odoo.com>
Co-authored-by: Alexis Lacroix (laa) <laa@odoo.com>
Co-authored-by: Lucas Lefèvre (lul) <lul@odoo.com>
Co-authored-by: Dhrutik Patel (dhrp) <dhrp@odoo.com>
Co-authored-by: Adrien Minne (adrm) <adrm@odoo.com>
Co-authored-by: Mehdi Rachico (mera) <mera@odoo.com>
Co-authored-by: Florian Damhaut (flda) <flda@odoo.com>
Co-authored-by: Rémi Rahir (rar) <rar@odoo.com>
Co-authored-by: Pierre Rousseau (pro) <pro@odoo.com>
Co-authored-by: Vincent Schippefilt (vsc) <vsc@odoo.com>

C3POdoo committed:
[I18N] *: export 17.0 source terms

C3POdoo committed:
[I18N] *: fetch 17.0 translations
